### DISCUSSION TITLE
Add support for URLRequest's `httpBodyStream`

### DISCUSSION THREAD

**Discussion by alexzagu** [pointfreeco/swift-snapshot-testing#945] [General]
Snapshotting an URLRequest instance after it has been processed by the `URL Loading System` (by subclassing `URLProtocol`) doesn't work properly at the moment, since the framework has support for `httpBody` but not for `httpBodyStream`. 

While processing the request, the `URL Loading System` places the data that was in the `httpBody` into the `httpBodyStream` before sending it to any subclass of `URLProtocol` to be fulfilled. That's why, if you're using a spy subclass to catch requests in tests, snapshotting won't reveal the data inside the request as of now.

**Comment by grdsdev** [comment]
I'm also in need of that, can you review this @mbrandonw or @stephencelis? I see linked PR has merge conflicts, I can fix that if you're willing to merge it.

I'm currently running a fork because of that https://github.com/grdsdev/swift-snapshot-testing

**Comment by mbrandonw** [comment]
Hi @grdsdev, we do not plan on merging that PR at this time. However, instead of depending on a fork, why don't you simply extract that snapshot strategy out of the library and make any changes you want to it? The strategy does not use any internals of the library, and so you should be able to maintain it yourself without involving a fork.

**Reply by grdsdev** [reply]
Thanks, @mbrandonw I will do that.

Just out of curiosity, why wouldn't this get merged at this time? Is it an issue with implementation?

**Reply by mbrandonw** [reply]
Hi @grdsdev, we have found time and time again that these snapshot strategies are very brittle. People will often submit innocent looking PR's to fix a small problem, only to find out it breaks someone else's snapshots.

So, we think the snapshot strategies that ship with this library should only be thought of as the basics to get you started. If eventually they do not suite your needs, then you should look into defining your own strategy. You can use ours as a starting point, and you can even open source it if you want to share it with others.
